Core/src/org/sleuthkit/autopsy/textextractors/SqliteTextExtractor.java

```diff
@@ -32,26 +32,62 @@
 import org.sleuthkit.datamodel.AbstractFile;
 
 /**
- * Dedicated SqliteTextExtractor to solve the problems associated with Tika's
- * Sqlite parser.
+ * Extracts text from SQLite database files.
  *
- * Tika problems: 1) Tika fails to open virtual tables 2) Tika fails to open
- * tables with spaces in table name 3) Tika fails to include the table names in
- * output (except for the first table it parses)
+ * This is a dedicated solution to address the problems associated with 
+ * Tika's sqlite parser (version 1.17), which include the following:
+ *  1) Virtual tables cause the parser to bail
+ *  2) Tables that contain spaces in their name are not extracted
+ *  3) Table names are not included in its output text
  */
-class SqliteTextExtractor extends ContentTextExtractor {
+public final class SqliteTextExtractor extends ContentTextExtractor {
 
     private static final String SQLITE_MIMETYPE = "application/x-sqlite3";
     private static final Logger logger = Logger.getLogger(SqliteTextExtractor.class.getName());
 
+    /**
+     * Accepts a context instance for run-time configuration.
+     * 
+     * As of now, this constructor is a no-op as it does not support 
+     * any type of configuration.
+     * 
+     * @param context Instance that contains config classes
+     */
+    public SqliteTextExtractor(ExtractionContext context) {
+    }
+    
+    /**
+     * Creates a default SqliteTextExtractor instance.
+     */
+    public SqliteTextExtractor() {
+    }
+
+    /**
+     * This extractor only works for sqlite files, so it is indeed content type
+     * specific. 
+     * 
+     * @return true
+     */
     @Override
-    boolean isContentTypeSpecific() {
+    public boolean isContentTypeSpecific() {
         return true;
     }
 
+    /**
+     * Determines if this extractor is fit to run.
+     * 
+     * @return Flag indicating if it should or shouldn't be run.
+     */
     @Override
     public boolean isDisabled() {
-        return false;
+        try {
```

Comment:
Is it possible for the SQLite JDBC class to be unavailable when this method is first called and subsequently be available at some point later when this method is called again?
If not I'd suggest only attempting to load the class once (and reduce the number of times the message appears in the logs). Does this really call for a SEVERE log message?

Reply:
I agree it shouldn't be done every time that the method is called. My next commit will have this fixed.